AzureFunction/Utilities.cs

```diff
@@ -0,0 +1,160 @@
+
+            string packageName = Regex.Match(policy, @"package\s([a-zA-Z0-9.]+)", RegexOptions.IgnoreCase).Groups[1].Value;
+            log.LogInformation(string.Format("Package name : {0}", packageName));
+
+
+            if (string.IsNullOrWhiteSpace(packageName))
+            {
+                outputLog = "No package name could be inferred from the policy. Cannot continue execution. Ensure that policy contains a package name defined";
+                taskLogger?.Log(outputLog).ConfigureAwait(false);
+
+                return new List<string> { outputLog };
+            }
+
+            string output;
+
+            try
+            {
+                File.WriteAllText(imageProvenancePath, imageProvenance);
+                log.LogInformation("Image provenance file created");
+                File.WriteAllText(policyFilePath, policy);
+                log.LogInformation("Policy content file created");
+
+                string arguments = string.Format(
+                            CultureInfo.InvariantCulture,
+                            "/c \"\"{0}\\opa_windows_amd64.exe\" eval -f pretty --explain notes -i \"{1}\\{2}\" -d \"{1}\\{3}\" \"data.{4}.violations\" > \"{1}\\{5}\"\"",
+                            executionContext.FunctionAppDirectory,
+                            folderName,
+                            ImageProvenanceFileName,
+                            PolicyFileName,
+                            packageName,
+                            OutputResultFileName);
+
+                var process = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = "cmd.exe",
+                        Arguments = arguments,
+                        UseShellExecute = false,
+                        RedirectStandardOutput = true,
+                        WorkingDirectory = executionContext.FunctionDirectory
+                    }
+                };
+
+                taskLogger?.Log("Initiating evaluation");
+                process.Start();
+                taskLogger?.Log("Evaluation is in progress");
+                process.WaitForExit();
+                taskLogger?.Log("Evaluation complete. Processing result");
+                log.LogInformation("Completed executing OPA");
+
+                output = File.ReadAllText(string.Format("{0}\\{1}", newFolderPath, OutputResultFileName));
+                log.LogInformation(output);
+            }
+            finally
+            {
+                Directory.Delete(newFolderPath, true);
+            }
+
+            return Utilities.GetViolationsFromResponse(log, taskLogger, output, out outputLog);
+        }
+
+        public static IEnumerable<string> GetViolationsFromResponse(
+        ILogger log,
+        TaskLogger taskLogger,
+        string output,
+        out string outputLog)
+        {
+            IEnumerable<string> violations;
+            int outputValueIndex = output.IndexOf("[\n");
```

Comment:
If we don't find "[]" then we need to check processExist code to see command succeeded or failed. if it failed then we need to read standard error and log that errors in violations list.

Reply:
Same as above. If violations is undefined. It is also an error